### PR TITLE
Per-tile transforms

### LINKED ISSUE #98: Per-tileset or per-tile transform
WGS84 is not appropriate for all tilesets.  For example, the inside of a building or a point cloud may be relative to local coordinate system.

This could be solved by adding a 4x4 `modelMatrix` (or `matrix` like glTF, or maybe `transform`) to `tileset.json`, which transforms the tileset's local coordinates to WGS84.  This would default to being the identity matrix, meaning that the local coordinate system is WGS84.

Alternatively, each tile in `tileset.json`could have a 4x4 `modelMatrix` that transforms to the parent's coordinate system (this would include the bounding volume), which provides more flexibility and is standard in the computer graphics community, but has a bit more complexity.  This is most likely the right approach to elegantly support heterogeneous tilesets of tilesets.

### PR DESCRIPTION
**Pull request by pjcozzi**
Fixes #98.

See
* [Coordinate System and Units](https://github.com/AnalyticalGraphicsInc/3d-tiles/tree/transform#coordinate-system-and-units)
* [Tile transform](https://github.com/AnalyticalGraphicsInc/3d-tiles/tree/transform#tile-transform)
* [External Tilesets](https://github.com/AnalyticalGraphicsInc/3d-tiles/tree/transform#external-tilesets)

The new figures will be updated later for consistency with the rest of the spec.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -162,10 +164,110 @@ The file extension of `content.url` defines the [tile format](#tileFormats).  Th
...
+        var child = children[k];
+        var childToRoot = defined(child.transform) ? Matrix4.fromArray(child.transform) : Matrix4.IDENTITY;
+        childToRoot = Matrix4.multiplyTransformation(transformToRoot, childToRoot, childToRoot);
+        stack.push({
+            tile : child,
+            transformToRoot : childToRoot
+        });        
+    }
+}
+```
```
Comment (lilleyse) [inline README.md]: I think a recursive example may be easier to grasp here for most readers.

**Comment by lilleyse** [comment]
Wow this covers a lot of small but important cases. Looks good to me!

File: README.md (hunk 1 above)
Comment (pjcozzi) [inline README.md]: Ah, yes, it is much shorter.  If you have a Cesium unit test handy, could you quickly run this?

Comment (pjcozzi) [inline README.md]: It is here: f557f562ad2777bc3173e8905ea9ff09c10d4f55

File: README.md [hunk 2]
```diff
@@ -162,10 +164,102 @@ The file extension of `content.url` defines the [tile format](#tileFormats).  Th
...
+    var children = tile.children;
+    var length = children.length;
+    for (var k = 0; k < length; ++k) {
+        var child = children[k];
+        var childToRoot = defined(child.transform) ? Matrix4.fromArray(child.transform) : Matrix4.IDENTITY;
+        childToRoot = Matrix4.multiplyTransformation(transformToRoot, childToRoot, childToRoot);
+        computeTransform(child, childToRoot);
+    }
+}
+```
```
Comment (lilleyse) [inline README.md]: Had to make two edits. Here's the working code.

```javacscript
        function computeTransforms(tileset) {
            var t = tileset.root;
            var transformToRoot = defined(t.transform) ? Matrix4.fromArray(t.transform) : Matrix4.IDENTITY;

            computeTransform(t, transformToRoot);
        }

        function computeTransform(tile, transformToRoot) {
            // Apply 4x4 transformToRoot to this tile's positions and bounding volumes

            var inverseTransform = Matrix4.inverse(transformToRoot, new Matrix4());
            var normalTransform = Matrix4.getRotation(inverseTransform, new Matrix3());
            normalTransform = Matrix3.transpose(normalTransform, normalTransform);
            // Apply 3x3 normalTransform to this tile's normals

            var children = tile.children;
            var length = children.length;
            for (var k = 0; k < length; ++k) {
                var child = children[k];
                var childToRoot = defined(child.transform) ? Matrix4.fromArray(child.transform) : Matrix4.clone(Matrix4.IDENTITY);
                childToRoot = Matrix4.multiplyTransformation(transformToRoot, childToRoot, childToRoot);
                computeTransform(child, childToRoot);
            }
        }
```

Comment (lilleyse) [inline README.md]: I'll just commit the change.

**Comment by pjcozzi** [comment]
Thanks @lilleyse!  Is this ready?

Does the spec need to cover anything else based on your implementation experience?

**Comment by lilleyse** [comment]
It covers everything I can think of.

**Pull request merged**